### PR TITLE
Migrate build scripts to newer patterns

### PR DESCRIPTION
**Pull request by tir38**
* Migrate all `build.gradle` [from Groovy to Kotlin](https://developer.android.com/build/migrate-to-kotlin-dsl)
* Migrate to [version catalog](https://developer.android.com/build/migrate-to-catalogs)

Gradle sync is happy. Sample app builds. I just need to make sure CI is happy with SDK

The reason for this is to help migrate to a new sample app (which already uses these changes)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,85 +0,0 @@
...
-
-    configurations {
-        ktlint
-    }
-
-    dependencies {
-        ktlint "com.pinterest:ktlint:${versions.ktlint}"
-    }
-
-    task ktlint(type: JavaExec, group: "verification") {
```
Comment (tir38) [inline build.gradle:68]: IMO we shouldn't have been defining these gradle tasks ourselves. We should have just used one of the many KtLint Gradle plugins

File: gradle.properties [hunk 2]
```diff
@@ -18,3 +18,5 @@ org.gradle.jvmargs=-Xmx2048m
 android.useAndroidX=true
 android.enableJetifier=true
 kotlin.code.style=official
+
+android.disableAutomaticComponentCreation=true
```
Comment (tir38) [inline gradle.properties:22]: >Software Components will not be created automatically for Maven publishing from Android Gradle Plugin 8.0. To opt-in to the future behavior, set the Gradle property android.disableAutomaticComponentCreation=true in the `gradle.properties` file or use the new publishing DSL.

File: afterpay/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+    buildTypes {
+        create("staging") {
+            initWith(getByName("debug"))
+        }
+    }
+
+    sourceSets {
+        val main by getting
+        main.java.srcDirs("src/main/kotlin")
```
Comment (Egorand) [inline afterpay/build.gradle.kts:31]: Doesn't the Kotlin plugin pick these sourcesets up automatically?

File: example/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    testOptions {
+        execution = "ANDROIDX_TEST_ORCHESTRATOR"
+    }
+
+    namespace = "com.example.afterpay"
+}
+
+tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>().all {
```
Comment (Egorand) [inline example/build.gradle.kts:64]: nit: import?

File: example/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    kotlinOptions.freeCompilerArgs += listOf("-Xallow-result-return-type")
+}
+
+dependencies {
+    implementation(project(":afterpay"))
```
Comment (Egorand) [inline example/build.gradle.kts:69]: nit: add `enableFeaturePreview("TYPESAFE_PROJECT_ACCESSORS")` to `settings.gradle.kts` so this becomes `projects.afterpay`?

File: gradle/libs.versions.toml [hunk 6]
```diff
@@ -0,0 +1,88 @@
+[versions]
+compileSdkVersion = "31"
+exampleCompileSdk = "33"
+minSdk = "24"
+java = "1.8"
+buildTools = "30.0.3"
+
+core = "2.3.0"
```
Comment (Egorand) [inline gradle/libs.versions.toml:8]: nit: `paykit`? `core` feels too generic, it's not clear what it refers to. Even better would be to just inline this version into the dependency declaration, usually there's no reason to declare version separately when it's only used by one dep:

```toml
[libraries]
paykit = "app.cash.paykit:core:2.3.0"
```

File: gradle/libs.versions.toml [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+coreVersion = "1.5.0"
+espressoCore = "3.5.1"
```
Comment (Egorand) [inline gradle/libs.versions.toml:10]: nit: `espresso`?

**Review by Egorand — APPROVED** [review]

File: afterpay/build.gradle.kts (hunk 3 above)
Comment (tir38) [reply afterpay/build.gradle.kts:31]: Probably.  Looks like everything still compiles w/ these removed.

**Review by ScottAntonacAP — APPROVED** [review]

**Pull request merged**